### PR TITLE
Refactor/14  Extracted file backends as separate repositories 

### PR DESCRIPTION
**Pull request by jakobbraun**
Cloese #14

### CODE REVIEW

File: NOTICE [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  TCP to Unix Socket Proxy under MIT
+  Test containers for Exasol on Docker under MIT
+  Test Database Builder for Java under MIT
+  Testcontainers :: Database-Commons under MIT
+  Testcontainers :: JDBC under MIT
+  Testcontainers :: JUnit Jupiter Extension under MIT
+  Testcontainers Core under MIT
+  Virtual Schema for document data in files under MIT
+  Visible Assertions under MIT
+
```
Comment (AnastasiiaSergienko) [inline NOTICE:64]: Can we divide them into runtime and test dependencies? And maybe direct and transitive dependencies too. This list is less informative than the manual one we had before

File: src/test/java/com/exasol/adapter/document/documentfetcher/files/CloseCheckStream.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.exasol.adapter.document.documentfetcher.files;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+class CloseCheckStream extends ByteArrayInputStream {
+    boolean wasClosed = false;
```
Comment (AnastasiiaSergienko) [inline src/test/java/com/exasol/adapter/document/documentfetcher/files/CloseCheckStream.java:7]: Should it be private?

**Review by AnastasiiaSergienko — APPROVED** [review]
Approved with one comment

File: src/test/java/com/exasol/adapter/document/documentfetcher/files/CloseCheckStream.java (hunk 2 above)
Comment (jakobbraun) [reply src/test/java/com/exasol/adapter/document/documentfetcher/files/CloseCheckStream.java]: :+1:

File: NOTICE (hunk 1 above)
Comment (jakobbraun) [reply NOTICE:64]: not yet.

**Pull request merged**